lib/src/presentation/widget/pokemon_data.dart

```diff
@@ -0,0 +1,146 @@
+                        style: TextStyles.detailsWeightAndHeightTextStyle,
+                      ),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.all(Constants.detailsPadding),
+                      child: Text(
+                        '${Constants.detailsHeightLabel} ${pokemonEntity.height}',
+                        style: TextStyles.detailsWeightAndHeightTextStyle,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.all(Constants.detailsPadding),
+              child: Container(
+                decoration: BoxDecoration(
+                  color: ColorsConstants.detailsAttributesBoxColor,
+                  border: Border.all(
+                    color: Colors.black,
+                    width: Constants.detailsBoxBorderWidth,
+                  ),
+                  borderRadius: BorderRadius.circular(
+                    Constants.detailsBoxBorderRadius,
+                  ),
+                ),
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    DetailBoxText(
+                      text:
+                          '${Constants.detailsBabyLabel} ${pokemonEntity.species!.isBaby ? Constants.yesValue : Constants.noValue}',
+                    ),
+                    DetailBoxText(
+                      text:
+                          '${Constants.detailsLegendaryLabel} ${pokemonEntity.species!.isLegendary ? Constants.yesValue : Constants.noValue}',
+                    ),
+                    DetailBoxText(
+                      text:
+                          '${Constants.detailsMythicalLabel} ${pokemonEntity.species!.isMythical ? Constants.yesValue : Constants.noValue}',
+                    ),
+                    DetailBoxText(
+                      text:
+                          '${Constants.detailsHabitatLabel} ${pokemonEntity.species?.habitat.name}',
+                    ),
+                    DetailBoxText(
+                      text:
+                          '${Constants.detailsShapeLabel} ${pokemonEntity.species?.shape.name}',
+                    ),
+                  ],
+                ),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.all(Constants.detailsPadding),
+              child: Text(
+                Constants.descriptionTitle,
+                style: TextStyles.detailsDescriptionTitleTextStyle,
+              ),
+            ),
+            Text(
+              '${pokemonEntity.species?.flavorTextEntries.first.flavorText}',
+              style: TextStyles.detailsDescriptionTextStyle,
+              textAlign: TextAlign.justify,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
```

Comment:
As an improvement you can make custom widget to divide this UI, like pokemonDescription, pokemonImage and something like that to make more shortest.

Reply:
refactorized